psiminer-core/src/main/kotlin/labelextractor/MethodCommentLabelExtractor.kt

```diff
@@ -1,29 +1,43 @@
 package labelextractor
 
-import astminer.common.splitToSubtokens
-import com.intellij.psi.PsiComment
 import com.intellij.psi.PsiElement
 import com.intellij.psi.javadoc.PsiDocComment
-import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.kdoc.psi.impl.KDocImpl
+import org.jetbrains.kotlin.util.capitalizeDecapitalize.toLowerCaseAsciiOnly
 import psi.language.LanguageHandler
 
-class MethodCommentLabelExtractor : LabelExtractor() {
+class MethodCommentLabelExtractor(private val onlyDoc: Boolean = true) : LabelExtractor() {
     override val granularityLevel = GranularityLevel.Method
 
     override fun handleTree(root: PsiElement, languageHandler: LanguageHandler): Label? {
-        val methodCommentNodes = PsiTreeUtil.collectElementsOfType(root, PsiComment::class.java)
 
-        val methodDocComments = methodCommentNodes.filterIsInstance<PsiDocComment>()
-            .map { it.descriptionElements }.flatMap { it.map { psiElement -> psiElement.text } }
-        val methodComments = methodCommentNodes.filter { it !is PsiDocComment }.map { it.text }
-        val onlyCommentsText = methodDocComments + methodComments
+        val docComment = languageHandler.methodProvider.getDocComment(root)
 
-        val filteredCommentLabel = onlyCommentsText.flatMap { splitToSubtokens(it) }
-            .filterNot { (it == "/*") || (it == "*/") || (it.none { c -> c.isLetterOrDigit() }) }.joinToString("|")
-        return if (filteredCommentLabel.isEmpty()) {
+        val nonAlphaNumericRegex = Regex("[^A-Za-z\\d]")
+
+        val docCommentText = if (docComment == null) {
+            ArrayList()
+        } else {
+            when (languageHandler.language) {
+                Language.Java -> {
+                    (docComment as PsiDocComment).descriptionElements.map { it.text }
+                        .flatMap { it.split(nonAlphaNumericRegex) }
+                }
+                Language.Kotlin -> {
+                    (docComment as KDocImpl).getDefaultSection().getContent().split(nonAlphaNumericRegex)
+                }
+            }
```

Comment:
This functionality is out of place here and should be moved to the language handler. The idea of language handlers is that they abstract all the details of language support implementation. In current implementation, your extractor has to know everything about each supported language which is not cool.

Create a function somewhere in the language handler hierarchy that will return you what you need.